src/main/java/org/folio/inventory/consortium/services/ConsortiumServiceImpl.java

```diff
@@ -0,0 +1,104 @@
+package org.folio.inventory.consortium.services;
+
+import io.vertx.core.Future;
+import io.vertx.core.http.HttpClient;
+import io.vertx.core.json.Json;
+import io.vertx.core.json.JsonArray;
+import io.vertx.ext.web.client.WebClient;
+import org.apache.http.HttpStatus;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.folio.inventory.common.Context;
+import org.folio.inventory.consortium.entities.ConsortiumConfiguration;
+import org.folio.inventory.consortium.entities.SharingInstance;
+import org.folio.inventory.consortium.entities.SharingStatus;
+import org.folio.inventory.consortium.exceptions.ConsortiumException;
+import org.folio.inventory.support.http.client.OkapiHttpClient;
+
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.util.Optional;
+import java.util.UUID;
+import java.util.concurrent.CompletableFuture;
+
+import static org.folio.inventory.dataimport.handlers.matching.util.EventHandlingUtil.constructContext;
+
+public class ConsortiumServiceImpl implements ConsortiumService {
+  private static final Logger LOGGER = LogManager.getLogger(ConsortiumServiceImpl.class);
+  private static final String USER_TENANTS_ENDPOINT = "/user-tenants?limit=1";
+  private static final String SHARE_INSTANCE_ENDPOINT = "/consortia/%s/sharing/instances";
+  private static final String SHARING_INSTANCE_ERROR = "Error during sharing Instance for sourceTenantId: %s, targetTenantId: %s, instanceIdentifier: %s, status code: %s, response message: %s";
+  private final HttpClient httpClient;
+
+  public ConsortiumServiceImpl(HttpClient httpClient) {
+    this.httpClient = httpClient;
+  }
+
+  @Override
+  public Future<SharingInstance> createShadowInstance(Context context, String instanceId, ConsortiumConfiguration consortiumConfiguration) {
+    Context centralTenantContext = constructContext(consortiumConfiguration.getCentralTenantId(), context.getToken(), context.getOkapiLocation());
+    SharingInstance sharingInstance = new SharingInstance();
+    sharingInstance.setSourceTenantId(consortiumConfiguration.getCentralTenantId());
+    sharingInstance.setInstanceIdentifier(UUID.fromString(instanceId));
+    sharingInstance.setTargetTenantId(context.getTenantId());
+    return shareInstance(centralTenantContext, consortiumConfiguration.getConsortiumId(), sharingInstance);
+  }
+
+  @Override
+  public Future<Optional<ConsortiumConfiguration>> getConsortiumConfiguration(Context context) {
+    CompletableFuture<Optional<ConsortiumConfiguration>> completableFuture = createOkapiHttpClient(context)
```

Comment:
Is there any way the okapi client can use vertx future instead of CompletableFuture?

Reply:
We cannot user vert.x future, as far as OkapiHttpClient uses CompletionStage on post/get requests